### PR TITLE
WFLY-17488 Create clustering subsystem/deployment parsers on demand.

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-17488

### CODE REVIEW

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/PersistentXMLDescriptionMarshaller.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import org.jboss.as.controller.PersistentResourceXMLDescription;
+import org.jboss.as.controller.persistence.SubsystemMarshallingContext;
+import org.jboss.dmr.ModelNode;
+import org.jboss.staxmapper.XMLElementReader;
+import org.jboss.staxmapper.XMLElementWriter;
+import org.jboss.staxmapper.XMLExtendedStreamWriter;
+
+/**
+ * A marshaller based on a {@link PersistentResourceXMLDescription} that is both an {@link XMLElementReader} and @link XMLElementWriter}.
```
Comment (rhusar) [inline clustering/common/src/main/java/org/jboss/as/clustering/controller/PersistentXMLDescriptionMarshaller.java:38]:
```suggestion
 * A marshaller based on a {@link PersistentResourceXMLDescription} that is both an {@link XMLElementReader} and {@link XMLElementWriter}.

**Review by rhusar — APPROVED** [review]
Looks good to me.

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/PersistentXMLDescriptionMarshaller.java (hunk 1 above)
Comment (pferraro) [reply clustering/common/src/main/java/org/jboss/as/clustering/controller/PersistentXMLDescriptionMarshaller.java:38]: Doh!

Comment (pferraro) [reply clustering/common/src/main/java/org/jboss/as/clustering/controller/PersistentXMLDescriptionMarshaller.java:38]: I'll patch that in when I merge.

**Pull request merged**